apps/promotion-app/backend/modules/database/utils.bal

```diff
@@ -38,3 +39,59 @@ isolated function buildSqlSelectQuery(sql:ParameterizedQuery mainQuery, sql:Para
 
     return updatedQuery;
 }
+
+# use for process user permissions.
+#
+# + permissions - user permission string
+# + return - Return Value Description
+public isolated function processPermissions(string? permissions) returns FunctionalLeadAccessLevels|error? {
+
+    // If the permissions string is nil, there's nothing to process; return unit `()`.
+    if permissions is () {
+        return;
+    }
+
+    // Attempt to parse the permission string as JSON.
+    json permissionJSON = check permissions.fromJsonString();
+
+    // Attempt to map the parsed JSON to a strongly-typed `FunctionalLeadAccessLevels` object.
+    return check permissionJSON.cloneWithType(FunctionalLeadAccessLevels);
+};
+
+# Helper function to user has roles.
+#
+# + requiredRole - Required Role list  
+# + userRoles - Roles list, the user has 
+# + return - boolean
+public isolated function checkRoles(Role[] requiredRole, Role[] userRoles) returns boolean {
```

Comment:
Couldn't find a usage for this function. Anyway, this doesn't has to be a separate function. Should be a one liner.

```ballerina
boolean hasRequiredRoles = requiredRoles.every(role => userRoles.indexOf(role) is int);
```

Parameter names are also not ideal. Since these are arrays, the name should be plural - `requiredRoles`